tools/src/test/java/org/apache/kafka/tools/consumer/group/ListConsumerGroupTest.java

```diff
@@ -20,78 +20,225 @@
+import org.junit.jupiter.params.provider.MethodSource;
 
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.EnumSet;
+import java.util.HashSet;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.Properties;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicReference;
 
+import static org.apache.kafka.tools.ToolsTestUtils.TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class ListConsumerGroupTest extends ConsumerGroupCommandTest {
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testListConsumerGroups(String quorum) throws Exception {
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
+    @MethodSource("getTestQuorumAndGroupProtocolParametersAll")
+    public void testListConsumerGroupsWithoutFilters(String quorum, String groupProtocol) throws Exception {
         String simpleGroup = "simple-group";
+
+        createOffsetsTopic(listenerName(), new Properties());
+
         addSimpleGroupExecutor(simpleGroup);
         addConsumerGroupExecutor(1);
+        addConsumerGroupExecutor(1, PROTOCOL_GROUP, groupProtocol);
 
         String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--list"};
         ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
-        scala.collection.Set<String> expectedGroups = set(Arrays.asList(GROUP, simpleGroup));
+
+        scala.collection.Set<String> expectedGroups = set(Arrays.asList(GROUP, simpleGroup, PROTOCOL_GROUP));
         final AtomicReference<scala.collection.Set> foundGroups = new AtomicReference<>();
+
         TestUtils.waitForCondition(() -> {
             foundGroups.set(service.listConsumerGroups().toSet());
             return Objects.equals(expectedGroups, foundGroups.get());
         }, "Expected --list to show groups " + expectedGroups + ", but found " + foundGroups.get() + ".");
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testListWithUnrecognizedNewConsumerOption() {
+    @Test
+    public void testListWithUnrecognizedNewConsumerOption() throws Exception {
         String[] cgcArgs = new String[]{"--new-consumer", "--bootstrap-server", bootstrapServers(listenerName()), "--list"};
         assertThrows(OptionException.class, () -> getConsumerGroupService(cgcArgs));
     }
 
-    @ParameterizedTest
-    @ValueSource(strings = {"zk", "kraft"})
-    public void testListConsumerGroupsWithStates() throws Exception {
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
+    @MethodSource("getTestQuorumAndGroupProtocolParametersAll")
+    public void testListConsumerGroupsWithStates(String quorum, String groupProtocol) throws Exception {
```

Comment:
I think we talked about this before, I thought we didn't want to make changes to the states in this PR so I had left it so we can improve it in another small PR, but I can change it now, I might've missed communicating that

Reply:
Sorry, I very likely missed your reply. It would be better to change it in my opinion. We could pass the group protocol to addConsumerGroupExecutor and adapt the assertions.

Reply:
okay thanks, I created two tests, we can't change the assertions without if statements and we didn't want those right